snapcraft/cli/remote.py

```diff
@@ -0,0 +1,245 @@
+    is complete, packages for each architecture are retrieved and will be available in
+    the local filesystem.
+
+    If the local files are under git revision control, use option --git to build the
+    current branch of this repository. Be aware, however, that untracked files and
+    uncommitted changes will not be used.
+
+    If not specified in the snapcraft.yaml file, the list of architectures to build
+    can be set using the --arch option. If both are specified, the architectures listed
+    in snapcraft.yaml take precedence.
+
+    Interrupted remote builds can be resumed using the --recover option, followed by
+    the build number informed when the remote build was originally dispatched. The
+    current state of the remote build for each architecture can be checked using the
+    --status option.
+
+    \b
+    Examples:
+        snapcraft remote-build
+        snapcraft remote-build --arch=all
+        snapcraft remote-build --git --arch=arm64,armhf
+        snapcraft remote-build --recover 47860738
+        snapcraft remote-build --status 47860738
+    """
+
+    project = get_project()
+
+    remote_dir = os.path.join(
+        BaseDirectory.save_data_path("snapcraft"),
+        "projects",
+        project.info.name,
+        "remote-build",
+    )
+
+    remote_info = InfoFile(os.path.join(remote_dir, "remote.yaml"))
+    remote_info.load()
+    provider = "launchpad"
+    if "id" in remote_info:
+        build_id = remote_info["id"]
+        if "provider" in remote_info:
+            provider = remote_info["provider"]
+    else:
+        build_id = "snapcraft-{}".format(uuid.uuid4().hex)
+        remote_info["id"] = build_id
+        remote_info["provider"] = provider
+        remote_info.save()
+
+    # TODO: change login strategy after launchpad infrastructure is ready
+    lp = LaunchpadClient(project, build_id)
+    lp.login(user)
+
+    if status:
+        # Show build status
+        lp.recover_build(status)
+        lp.show_build_status()
+        return
+    elif recover:
+        # Recover from interrupted build
+        echo.info("Recover build {}...".format(recover))
+        lp.recover_build(recover)
+    else:
+        # If build architectures not set in snapcraft.yaml, let the user override
+        # Launchpad defaults using --arch.
+        archs = _list_architectures(project, arch)
+
+        # Sanity check for build architectures
+        _check_supported_archs(archs)
+
+        # The default branch to build
+        branch = "master"
+
+        # Send local data to the remote repository
+        echo.info("Sending data to remote builder...")
+        if git:
+            url, branch = _send_current_tree(provider, lp.user, build_id + "-git")
+        else:
+            url = _copy_and_send_tree(
+                provider, lp.user, remote_dir, build_id, project.info.name
+            )
+
+        # Create build recipe
+        lp.delete_snap()
```

Comment:
Can you comment on why we are deleting first?

Reply:
Add comment to code (delete any existing snap to remove leftovers from previous builds).